modules/linear_boltzmann_solvers/discrete_ordinates_problem/sweep/communicators/aah_async_comm.cc

```diff
@@ -214,18 +145,19 @@ AAH_ASynchronousCommunicator::ReceiveDelayedData(int angle_set_num)
 
   const auto& spds = fluds_.GetSPDS();
   const auto& comm = comm_set_.LocICommunicator(opensn::mpi_comm.rank());
-  const size_t num_delayed_dependencies = spds.GetDelayedLocationDependencies().size();
+  const std::size_t num_delayed_dependencies = spds.GetDelayedLocationDependencies().size();
 
   bool all_messages_received = true;
-  for (size_t i = 0; i < num_delayed_dependencies; ++i)
+  for (std::size_t i = 0; i < num_delayed_dependencies; ++i)
   {
     auto& upstream_psi = fluds_.DelayedPrelocIOutgoingPsi()[i];
 
-    assert(delayed_preloc_msg_data_[i].size() <= std::numeric_limits<int>::max());
-    for (int m = 0; m < delayed_preloc_msg_data_[i].size(); ++m)
+    for (std::size_t m = 0; m < delayed_preloc_msg_data_[i].size(); ++m)
     {
       const auto& [source, size, block_pos] = delayed_preloc_msg_data_[i][m];
-      const int tag = max_num_messages_ * angle_set_num + m;
+      const std::size_t utag = max_num_messages_ * static_cast<std::size_t>(angle_set_num) + m;
```

Comment:
`tag` was changed to `int`, because MPI uses tags as `int`, so probably should stay as `int`, unless you prefer those `static_cast`s...